docs-next/pages/apis/hooks.mdx

```diff
@@ -2,6 +2,455 @@ import { Markdown } from '../../components/Page';
+
+For field hooks, the return value should be an updated value for that specific field.
+For list hooks, the return value should be a [`resolved data`](#resolved-data-stages) object.
+The result of `resolveInput` will be passed as `resolvedData` into the next stages of the operation.
+
+| Argument        | Description                                                                                                                                                                       |
+| :-------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `listKey`       | The key of the list being operated on.                                                                                                                                            |
+| `fieldPath`     | The path of the field being operated on (field hooks only).                                                                                                                       |
+| `operation`     | The operation being performed (`'create'` or `'update'`).                                                                                                                         |
+| `originalInput` | The value of `data` passed into the mutation.                                                                                                                                     |
+| `existingItem`  | The currently stored item (`undefined` for `create` operations). This object is an unresolved list item. [list item API](./list-items) for more details on unresolved list items. |
+| `resolvedData`  | A [`resolved data`](#resolved-data-stages) object. The resolved data value after default values, relationship resolvers, and field resolvers have been applied.                   |
+| `context`       | The [`KeystoneContext`](./context) object of the originating GraphQL operation.                                                                                                   |
+
+```typescript
+import { config, createSchema, list } from '@keystone-next/keystone/schema';
+import { text } from '@keystone-next/fields';
+
+export default config({
+  lists: createSchema({
+    ListName: list({
+      hooks: {
+        validateInput: async ({
+          listKey,
+          operation,
+          originalInput,
+          existingItem,
+          resolvedData,
+          context,
+        }) => {
+          /* ... */
+          return resolvedData;
+        },
+      },
+      fields: {
+        fieldName: text({
+          hooks: {
+            validateInput: async ({
+              listKey,
+              fieldPath,
+              operation,
+              originalInput,
+              existingItem,
+              resolvedData,
+              context,
+            }) => {
+              /* ... */
+              return resolvedData[fieldName];
```

Comment:
I'm unclear here as to what happens when the field input isn't 1:1 with the field path..?

@mitchellhamilton has a proposal on this, to make fields and input 1:1, which I think helps clarify this in the best way, but that's a thing for later

Reply:
Which part is unclear? Could you give an example of an input and which arguments could lead to confusion?

Reply:
> I'm unclear here as to what happens when the field input isn't 1:1 with the field path..?

That doesn't/sort of can't happen for create/update.